spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -159,7 +160,15 @@ public Table loadTable(Identifier ident, String version) throws NoSuchTableExcep
           sparkTable.snapshotId() == null,
           "Cannot do time-travel based on both table identifier and AS OF");
 
-      return sparkTable.copyWithSnapshotId(Long.parseLong(version));
+      try {
+        return sparkTable.copyWithSnapshotId(Long.parseLong(version));
+      } catch (NumberFormatException e) {
+        SnapshotRef ref = sparkTable.table().refs().get(version);
+        ValidationException.check(
+            ref != null,
+            "Cannot find matching snapshot ID or reference name for version " + version);
+        return sparkTable.copyWithSnapshotId(ref.snapshotId());
```

Comment:
It always use the latest commit from the reference. 
I think we also need to provide a way to time travel to a snapshot within a branch/tag? 

So along with existing of `FOR SYSTEM_VERSION AS OF snapshotId`
we should have `FOR SYSTEM_VERSION AS OF snapshotId@refName`

But whether to use '@' or some other syntax is an open point for a long time which @rdblue wanted to conclude.  

Nessie SQL syntax for reference: 
https://projectnessie.org/tools/sql/#grammar

Reply:
Never mind. 
After thinking a bit more about it and reviewing #6573, As the snapshot log contains all the snapshots from all the branches/tags. If we want to use any particular snapshot, we can directly use snapshot-id without specifying branch/tag information. So, no need of `snapshotId@refName` syntax

Reply:
I think one thing that might be useful is to time travel in a branch, something like `FOR SYSTEM_VERSION AS OF branchName@123456789012`. But that feels very hacky, I' rather have some syntax as we have been suggesting like `SELECT * FROM table BRANCH branch FOR SYSTEM_TIME AS OF xxx`. So I am leaving that part out of the implementation for now. At least I think most people can agree that a tag/branch head can be viewed as a version to travel to.